### PR TITLE
[AppBar] Add animated tab switching example

### PR DESCRIPTION
**Pull request by brianjmoore**
Test case demonstrating issues with MDCFlexibleHeader and animated tab transitions in MDCAppBar.

Original issue:
https://github.com/material-components/material-components-ios/issues/6437

### CODE REVIEW

File: components/AppBar/examples/AppBarAnimatedGlitchExample.swift [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import CoreGraphics
+import UIKit
+import third_party_objective_c_material_components_ios_components_AppBar_AppBar
```
Comment (romoore) [inline components/AppBar/examples/AppBarAnimatedGlitchExample.swift]: These imports look like they were rewritten by the `.kokoro` script. It's best to have them start with something like:

```swift
import MaterialComponents.AppBar
import MaterialComponents.AppBar_ColorThemer
```

Comment (brianjmoore) [reply components/AppBar/examples/AppBarAnimatedGlitchExample.swift]: Good catch, done.

File: components/AppBar/examples/AppBarAnimatedGlitchExample.swift [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+  lazy var tabBar: MDCTabBar = {
+    let tabBar = MDCTabBar()
+
+    tabBar.items = self.tabs.map { $0.tabBarItem }
+
+    tabBar.delegate = self
+    return tabBar
+  }()
+
+  override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: Bundle?) {
```
Comment (romoore) [inline components/AppBar/examples/AppBarAnimatedGlitchExample.swift:49]: Does this get called from `-init`? CatalogByConvention doesn't call this initializer directly, it only calls `-init`.

File: components/AppBar/examples/AppBarAnimatedGlitchExample.swift [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+  }
+
+  override func viewDidLoad() {
+    super.viewDidLoad()
+
+    for (offset, tab) in zip(0..., tabs) {
+      tab.tabBarItem.tag = offset
+    }
+
+    view.backgroundColor = .white
```
Comment (romoore) [inline components/AppBar/examples/AppBarAnimatedGlitchExample.swift]: Tip: This can be removed since you reassign it below.

**Review by romoore — APPROVED** [review]

File: components/AppBar/examples/AppBarAnimatedGlitchExample.swift (hunk 2 above)
Comment (brianjmoore) [reply components/AppBar/examples/AppBarAnimatedGlitchExample.swift:49]: Yes, UIViewController.init calls through to `init(nibName:bundle:)`. As the designated initializer it's the preferred override point.

File: components/AppBar/examples/AppBarAnimatedGlitchExample.swift (hunk 3 above)
Comment (brianjmoore) [reply components/AppBar/examples/AppBarAnimatedGlitchExample.swift]: Good point, thx.

**Pull request merged**